crates/ra_syntax/src/ptr.rs

```diff
@@ -15,9 +15,8 @@ impl SyntaxNodePtr {
         SyntaxNodePtr { range: node.text_range(), kind: node.kind() }
     }
 
-    pub fn to_node(self, root: &SyntaxNode) -> SyntaxNode {
-        assert!(root.parent().is_none());
```

Comment:
I had to remove this so I could go from the function source and expr syntax pointer to the expr source. It doesn't seem to be a necessary prerequisite?

Reply:
Yeah, I think that's fine.